### ISSUE TITLE
no-undef: 'arguments' is undefined when used inside function parameters destructuring assignment

### ISSUE THREAD

**Issue by tophf** [eslint/eslint#10639]
* **ESLint Version:** 5.2.0
* **Node Version:** 10.6.0
* **npm Version:** 6.1.0

<details>
<summary>Configuration</summary>

```js
{
    "parserOptions": {
        "ecmaVersion": 6,
        "sourceType": "script",
        "ecmaFeatures": {}
    },
    "rules": {
        "no-undef": 2
    },
    "env": {}
}
```

</details>

```js
(function test({url = arguments[0]} = {}) {
  return url;
})('foo')
```

Expected: no errors
Observed: `'arguments' is not defined. (no-undef)`

P.S. `(function test({url} = {}) { return arguments[0] })()` is ok.

**Comment by tophf** [comment]
It is available.

**Comment by ljharb** [comment]
Are you verifying this with babel, or with a native implementation?

**Comment by ljharb** [comment]
Although now that I think about it, the arguments object can be fully formed prior to any of the functions default arguments being evaluated. So perhaps this is indeed an a bug.

**Comment by tophf** [comment]
Native implementation both in node's repl and in a browser's devtools.

**Comment by ljharb** [comment]
This is still a bug, and should be reopened.
